app/javascript/components/domain-form/index.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './domain-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = (id, f) => {
+  miqSparkleOn();
+  API.get(`/api/automate_domains/${id}?attributes=name,description,enabled,source`).then(({
```

Comment:
This promise should be returned so you can use `then`/`catch` from the initial call.